agenda/presentation/src/main/java/com/pronoidsoftware/agenda/presentation/detail/AgendaDetailViewModel.kt

```diff
@@ -178,6 +296,50 @@ class AgendaDetailViewModel @Inject constructor(
                 )
             }
 
+            AgendaDetailAction.OnAllVisitorsClick -> {
+                state = state.copy(
+                    selectedVisitorFilter = VisitorFilterType.ALL,
+                )
+            }
+
+            AgendaDetailAction.OnGoingVisitorsClick -> {
+                state = state.copy(
+                    selectedVisitorFilter = VisitorFilterType.GOING,
+                )
+            }
+
+            AgendaDetailAction.OnNotGoingVisitorsClick -> {
+                state = state.copy(
+                    selectedVisitorFilter = VisitorFilterType.NOT_GOING,
+                )
+            }
+
+            AgendaDetailAction.OnToggleAddVisitorDialog -> {
+                state = state.copy(
+                    visitorToAddEmail = TextFieldState(),
+                    isShowingAddVisitorDialog = !state.isShowingAddVisitorDialog,
+                )
+            }
+
+            is AgendaDetailAction.OnAddVisitorClick -> {
+                viewModelScope.launch {
+                    state = state.copy(
+                        isAddingVisitor = true,
+                    )
+                    state = state.copy(
+                        visitorToAddEmail = TextFieldState(),
+                        isAddingVisitor = false,
+                        isShowingAddVisitorDialog = false,
+                    )
```

Comment:
I'm strongly assuming there's a reason you're doing two state updates here. Is there anything specific the switch of `isAddingVisitor` triggers? To show a progress bar, this will probably not be noticeable since there's no delay in between

Reply:
yeah, I should have left a TODO here, but between setting `isAddingVisitor` to true and then back to false, I was going to do the network call and make the "Add" button disabled with a circular progress bar so it can't be spammed. I also need to add here what happens if the API returns an error so that the message is shown and the dialog remains on screen